src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -1696,7 +1696,12 @@ private static ResourceBundle getBundleImpl(Module callerModule,
         Reference.reachabilityFence(callerModule);
         Reference.reachabilityFence(module);
 
-        return bundle;
+        //The OpenJ9 GC might collect the loader before we return here. This prevents that.
+        try {
+            return bundle;
+        } finally {
+            Reference.reachabilityFence(loader);
```

Comment:
Seems the reachabilityFence could be added above in the same place where its used for the modules, not sure there is any benefit to using try finally around the return statement.